kaluga-library-components/build.gradle.kts

```diff
@@ -0,0 +1,36 @@
+import org.jetbrains.kotlin.konan.file.File
+import org.jetbrains.kotlin.konan.properties.loadProperties
+
+plugins {
+    `java-gradle-plugin`
+    `kotlin-dsl`
+    `kotlin-dsl-precompiled-script-plugins`
+    `maven-publish`
+}
+
+repositories {
+    gradlePluginPortal() // To use 'maven-publish' and 'signing' plugins in our own plugin
+    google()
+    mavenCentral()
+}
+
+gradlePlugin {
+    plugins.register("kaluga-library-components") {
+        id = "kaluga-library-components"
+        implementationClass = "LibraryComponentsPlugin"
+    }
+}
+
+dependencies {
+
+    val properties = File("${rootDir.absolutePath}/../gradle.properties").loadProperties()
+    val kotlinVersion = properties["kaluga.kotlinVersion"] as String
+    val androidGradleVersion = properties["kaluga.androidGradlePluginVersion"] as String
+    val ktLintVersion = properties["kaluga.ktLintGradlePluginVersion"] as String
```

Comment:
So its a problem of build order and access:
I need these versions in both the component build as well as in the settings/build.gradle of the root lib (due to things like classpath and dependency resolution). Composite builds are compiled within their own separate thing, so the order of compilation is:
1. settings.gradle
2. composite builds
4. build.gradle (note that the code in buildScript doesnt yet have access to composite build because it doesnt apply a plugin yet)
5. module/build.gradle

Since I cant pass a version to the composite build there's no way to define a version that is accessible from both settings.gradle and kaluga-library-components/build.gradle.

Furthermore, the classess defined in kaluga-library-component dont have access to any properties file that is part of the composite build afaik, so aligning versions between Dependencies and build.gradle is also tricky.

I tried alternatives, but for now this is the best I could think of

Reply:
we can probably give it a try with version catalogs which we were planning to do later (they can share files and are loaded at settings time AFAIK), I'll resolve this for now.